tsconfig.json

```diff
@@ -1,12 +1,14 @@
 {
   "exclude": [
-    "test/*"
+    "test/*",
+    "tasks/*",
+    "gulpfile.ts"
   ],
   "compilerOptions": {
     "target": "ES2016",                       /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017','ES2018' or 'ESNEXT'. */
     "module": "commonjs",                     /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', or 'ESNext'. */
     "allowJs": true,                          /* Allow javascript files to be compiled. */
-    "outDir": "./dist",                       /* Redirect output structure to the directory. */
+    "outDir": "./dist/lib",                   /* Redirect output structure to the directory. */
```

Comment:
This seems to be causing me issues. When I compile and run, I get `Error: Cannot find module 'C:\Users\Daniel\Documents\GitHub\xud\dist\lib\Xud.js'`. The resulting directory structure is `dist/lib/lib/...` which seems wrong. If I revert this line, I get `dist/lib/...` which works. I think this may also affect loading of the `lnrpc.proto` file if it is changed. Can we simply leave this line as is?

Reply:
I guess it's a cross os compatibility issue, because before I added this change the compilation output structure was flat, meaning `Xud.js` location was `/dist/Xud.js` instead of `/dist/lib/Xud.js`.
This was introduced after I excluded `/tasks/*` and `gulpfile.ts` from compilation, which ended in `lib` being the only directory being compiled. 

So in order to remove this line we should either find a solution to enforce the `outDir` keep it's nested structure, or to just keep some unnecessary file/s to be compiled along with `lib`.

Reply:
I'm getting a `typedoc` folder within `dist` along with `lib`. Odd, I'm going to test this on linux as well and do some research.

Reply:
I didn't get this issue with linux. I did a lot of digging and the only way I could make this issue go away on Windows was by changing `allowJs` to false. Which is weird, because there are no `.js` files in my `lib` folder. I'll open an issue with the typescript project. But for the time being, I don't think we want to be compiling any `.js` files anyway, so maybe just leave this line as is and change `allowJs` to false so it doesn't cause problems on Windows.

Reply:
Actually, another way to fix this was to change the `excludes` option to `includes` and only specifying the lib folder. So maybe we want to use the approach of including only the folders we want, rather than excluding folders we don't want. Either way, I'm fine with changing `allowJs` to false since I don't think we'll be using that.